web/main.ts

```diff
@@ -2625,6 +2682,38 @@ class GitGraphView {
 				this.changeFileViewType(GG.FileViewType.List);
 			});
 
+			document.getElementById('cdvChooseParent')?.addEventListener('click', (event) => {
+				// Prevent closing of context menu by the same click
+				event.stopPropagation();
+
+				const expandedCommit = this.expandedCommit!;
+				const currentParentIndex = expandedCommit.parentIndex;
+				const parents = expandedCommit.commitDetails!.parents;
+
+				const contextMenuItems = parents.map((parent, index) => {
+					const parentIndex = index + 1;
+					const parentCommit = this.commitLookup[parent] !== undefined ? this.commits[this.commitLookup[parent]] : undefined;
+					const subject = parentCommit?.message.split('\n')[0];
+
+					return {

```

Comment:
I think this context menu could be improved by leveraging the "checked" context menu capability. This would allow the context menu to always contain an item for each parent, and there would be a check mark next to the current parent. For reference, the commit order items on the Table Column Header Context menu use this pattern.

This just requires the second attribute of `contextMenu.show` to be true, and a checked property to be added to each context menu item (`checked: parentIndex === currentParentIndex`).